app/main.py

```diff
@@ -1,8 +1,25 @@
 class Person:
-    # write your code here
-    pass
+
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.__class__.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    people_instances = []
+
+    for person in people:
+        person_instance = Person(person["name"], person["age"])
+        people_instances.append(person_instance)
+
+    for person in people:
+        class_people_person_name = Person.people[person["name"]]
+        if "wife" in person and person["wife"] is not None:
+            class_people_person_name.wife = Person.people[person["wife"]]
+        elif "husband" in person and person["husband"] is not None:
+            class_people_person_name.husband = Person.people[person["husband"]]
+
+    return people_instances
```

Comment:
You can return attribute people of Person's class

Reply:
But how? We have a dict as class attribute not a list -> "This class also should have a class attribute people, it is a dict that stores Person instances by their name"

Reply:
return list(Person.people.values())

Reply:
I see, thank you!